pkg/scheduler/objects/application_state.go

```diff
@@ -143,6 +143,9 @@ func NewAppState() *fsm.FSM {
 			},
 			fmt.Sprintf("enter_%s", Starting.String()): func(event *fsm.Event) {
 				app := event.Args[0].(*Application) //nolint:errcheck
+				if app.queue != nil {
```

Comment:
The app must have a queue set if it is in starting state, there is no need for a nil check